pages/items/new.tsx

```diff
@@ -27,20 +30,65 @@ fetch('http://localhost:3000/api/attributes', {
   })
 })
 
-export default function NewItemPage() {
+let itemDefinitionFormData: ItemDefinitionFormData
+
+async function createItem(formData: ItemDefinitionFormData) {
+  const itemDefReq = itemDefinitionFormDataToItemDefinitionRequest(formData)
+
+  const response = await fetch('http://localhost:3000/api/itemDefinitions', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(itemDefReq),
+  })
+
+  const data = await response.json()
+
+  return data.payload
+}
+
+interface Props {
+  backHref?: string
+}
+
+export default function NewItemPage({backHref}: Props) {
   const router = useRouter()
 
+  const redirectBack = (queryStr?: string) => {
+    if (backHref) {
+      router.push(`${backHref}${queryStr}`)
+    } else {
+      router.reload()
+
+    }
+  }
+
   return (
     <>
       <DialogTitle>Create New Item</DialogTitle>
       <DialogContent>
-        <UpsertItemForm categories={categories} attributes={attributes} />
+        <UpsertItemForm
+          categories={categories}
+          attributes={attributes}
+          onChange={(formData) => {
+            itemDefinitionFormData = formData
+          }}
+        />
       </DialogContent>
       <DialogActions>
-        <Button onClick={() => router.back()} color="inherit">
+        <Button onClick={() => redirectBack()} color="inherit">
           Close
         </Button>
```

Comment:
This doesn't seem to work properly
If I click "close" it sends me to "http://localhost:3000/checkInundefined" which gives error 404